float32 x_pos
float32 y_pos
float32 x_velocity_val
float32 y_velocity_val
